presto-main/src/main/java/io/prestosql/operator/PagesIndex.java

```diff
@@ -218,6 +220,9 @@ public void addPage(Page page)
 
         for (int position = 0; position < page.getPositionCount(); position++) {
             long sliceAddress = encodeSyntheticAddress(pageIndex, position);
+            if (valueAddresses.size() >= 2_000_000_000) {
```

Comment:
Should we use https://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#MIN_VALUE instead?

Reply:
@sopel39  `valueAddresses.size()` is an `int`, so surely kess than Integer.MAX_VALUE

@electrum where does 2B come from? safety measure?
please add a code comment

Reply:
I was picking a round number, like we do in other places (search for "1 billion")

Reply:
Added a comment